### PR TITLE
Fix error message in `scipy_match_demand`

### PR DESCRIPTION
**Pull request by tsmbland**
I just realised that one of the changes I made in #381 broke an error message in the `scipy_match_demand` function. This fixes it

### CODE REVIEW

**Comment by tsmbland** [comment]
Or not...

**Comment by dalonsoa** [comment]
I'm not sure I understand. Except for the `xfail` test, that PR was passing all checks, so what was broken?

**Comment by tsmbland** [comment]
It's [this error message](https://github.com/EnergySystemsModellingLab/MUSE_OS/blob/06d35aff4de6c31dff44887ca01cb43f09a7b682/src/muse/investments.py#L345), which is raised when the solver can't find a solution. It's not part of the testing suite, so I only noticed when dealing with an infeasible model. Obviously my first attempt to fix it hasn't worked (although strangely it works for me locally), so I'll try something else

**Comment by tsmbland** [comment]
Ok now it looks better

File: src/muse/investments.py [hunk 1]
```diff
@@ -310,8 +310,6 @@ def scipy_match_demand(
 
     from muse.constraints import ScipyAdapter
 
-    df_technologies = technologies.to_dataframe()
```
Comment (dalonsoa) [inline src/muse/investments.py:313]: So... is this not needed anymore?

**Review by dalonsoa — APPROVED** [review]
Assuming the failing windows check isn't relevant, it looks good!

File: src/muse/investments.py (hunk 1 above)
Comment (tsmbland) [reply src/muse/investments.py:313]: Nope. Previously it was converting the xarray to a dataframe and then getting the list of technologies from the dataframe, but you can get this directly from the xarray

**Pull request merged**